pkg/reconciler/testing/v1/rolebinding.go

```diff
@@ -17,27 +17,84 @@ limitations under the License.
 package testing
 
 import (
-	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func CreateRoleBinding(namespace, roleName string, sa *corev1.ServiceAccount) *v1.RoleBinding {
-	return &v1.RoleBinding{
+// RoleBindingOption enables further configuration of a RoleBinding.
+type RoleBindingOption func(*v1.RoleBinding)
+type SubjectOption func(*v1.Subject)
+type RoleRefOption func(*v1.RoleRef)
+
+func CreateRoleBinding(roleName, namespace string, o ...RoleBindingOption) *v1.RoleBinding {
+	rb := &v1.RoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      roleName + "-binding",
 			Namespace: namespace,
 		},
-		Subjects: []v1.Subject{
-			{
-				Kind: "ServiceAccount",
-				Name: sa.Name,
-			},
-		},
-		RoleRef: v1.RoleRef{
-			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "Role",
-			Name:     roleName,
-		},
+		Subjects: []v1.Subject{},
+		RoleRef:  v1.RoleRef{},
+	}
+	for _, opt := range o {
+		opt(rb)
+	}
+	return rb
+}
+
+func WithRoleBindingSubjects(subject ...v1.Subject) RoleBindingOption {
+	return func(rb *v1.RoleBinding) {
+		rb.Subjects = subject
+	}
+}
```

Comment:
```suggestion

func WithRoleBindingSubjects(subject ...*v1.Subject) RoleBindingOption {
	return func(rb *v1.RoleBinding) {
		rb.Subjects = *subject
	}
}
```
This should also help avoid having pointers in the test code